IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -286,4 +286,57 @@ - (MSIDAccountMetadataCacheItem *)retrieveAccountMetadataCacheItemForClientId:(N
     return cacheItem;
 }
 
+#pragma mark - Broker Utility
+
+// Remove account metadata for all clients based on home account id
+- (BOOL)removeAccountMetadataForHomeAccountId:(NSString *)homeAccountId
+                                      context:(id<MSIDRequestContext>)context
+                                        error:(NSError **)error
+{
+    if ([NSString msidIsStringNilOrBlank:homeAccountId])
+    {
+        if (error) *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"HomeAccountId is needed to remove account metadata!", nil, nil, nil, context.correlationId, nil, YES);
+        return NO;
+    }
+    
+    NSError *localError;
+    NSArray<MSIDAccountMetadataCacheItem *> *cacheItems = [self allAccountMetadataCacheItemsWithContext:context error:&localError];
+    if (localError)
+    {
+        if (error) *error = localError;
+        return NO;
+    }
+    
+    for (MSIDAccountMetadataCacheItem *cacheItem in cacheItems)
+    {
+        localError = nil;
+        [cacheItem removeAccountMetadataForHomeAccountId:homeAccountId error:&localError];
+        if (localError)
+        {
+            if (error) *error = localError;
```

Comment:
If there are multiple items and we fail to remove one of them, shouldn't we still continue the loop to remove other items if we can. We can have a boolean variable outside the loop and set it inside the loop incase of error and return that instead

Reply:
+1

Reply:
Sure let’s change that :)

Reply:
addressed.